lib/ArtistClass.ts

```diff
@@ -27,6 +39,8 @@ export default class Artist {
     this.location = location;
     this.slug = stringify(artistName);
     this.tattoos = tatoos;
-    this._id = null;
+    this.id = nanoid();
   }
```

Comment:
I saw that you already set up a database to manage artists. MongoDB creates `_id` values automatically so normally there should be no need to create ids on the client side with nanoid. I think you could have `id: null` as long as your artist hasn't been saved to the database. Once you save it, your `POST` route should return the `_id` value created by MongoDB and you can update your data accordingly.

Reply:
Tough one! My solution was to change the ArtistClass but keep the ArtisInterface. Since I don't need the _id-property during creation, but i do need it to render the pages.

```typescript
interface ArtistInterface {
  artistName: string;
  firstName: string;
  lastName: string;
  location: string;
  slug: string;
  tattoos: string[];
  _id: string;
}

class Artist {
  artistName: string;
  firstName: string;
  lastName: string;
  location: string;
  slug: string;
  tattoos: string[];

  constructor(
    artistName: string,
    firstName: string,
    lastName: string,
    location: string,
    tatoos: string[]
  ) {
    this.artistName = artistName;
    this.firstName = firstName;
    this.lastName = lastName;
    this.location = location;
    this.slug = stringify(artistName);
    this.tattoos = tatoos;
  }
}

export { Artist, type ArtistInterface };
```

Reply:
Hmm - But now your class does not match your interface. Couldn't you have `_id` as class property and just initialize it with `null` (to indicate a "not saved yet" state)?

Reply:
The problem ist, i need the _id property to render my page (/pages/index.tsx) If I set it to null, I get an error when trying to access the _id-property of my artist-object @index.tsx